belgif-rest-problem-openapi-validation-spring/src/main/java/io/github/belgif/rest/problem/openapi/OpenapiRequestViolationProblemAdvice.java

```diff
@@ -0,0 +1,181 @@
+package io.github.belgif.rest.problem.openapi;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import com.atlassian.oai.validator.report.ValidationReport;
+import com.atlassian.oai.validator.springmvc.InvalidRequestException;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.github.belgif.rest.problem.BadRequestProblem;
+import io.github.belgif.rest.problem.ResourceNotFoundProblem;
+import io.github.belgif.rest.problem.api.InEnum;
+import io.github.belgif.rest.problem.api.InputValidationIssue;
+import io.github.belgif.rest.problem.api.Problem;
+import io.github.belgif.rest.problem.spring.ProblemMediaType;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+@RestControllerAdvice
+@ConditionalOnWebApplication
+public class OpenapiRequestViolationProblemAdvice {
```

Comment:
I wonder if it can't be added directly to belgif-rest-problem-spring, with
@ConditionalOnClass(InvalidRequestException.class)?

Reply:
I'll try

Reply:
Seems to work fine.
Only difference between SB2 and 3 is that when a non-existing path is called, in SB3, this can be validated by the atlassian lib and catched by the controlleradvice, where in sb2, the exception is already thrown in the dispatcherServlet, and cannot be catched by the controlleradvice. When the atlassian lib is configured to allow unknown paths, spring handles the 404.